### DISCUSSION TITLE
Remove db \ Prisma from trpc context | Greatly increase auto complete speed.

### DISCUSSION THREAD

**Discussion by TheHinky** [t3-oss/create-t3-turbo#997] [Ideas]
### Describe the feature you'd like to request

many X autocomplete speed increase with large db/prisma schema types.

my typescript autocomplete was stupidly slow, like 30-40sec for one recommendation. 
Ive been searching for solutions for ages, I think I might have found it, but I need to test more and overall, Iid classify myself as still very new to web dev, so I wont create an issue or pr. 

### Describe the solution you'd like to see

My solution is simple, remove db/prisma from context of trpc. 
More detail explanation: 
https://github.com/trpc/trpc/discussions/5636#discussioncomment-9139969

My autocomplete have gone from 30-40secs to legitimately instant. 

I don't know any down sides yet, hence why I'm posting here to see if anyone could tell me why this is horrible idea or if this should be recommended in t3 turbo. 

### Additional information

Again i am very new to all of this, and just posting to see if it might greatly help people in similar situation to me.

**Comment by noxify** [comment]
is this also an issue with drizzle or just with prisma?

To your solution: 
You have deleted `db` here: https://github.com/t3-oss/create-t3-turbo/blob/main/packages/api/src/trpc.ts#L39

And if you need access to the database, you import the db package and call it directly instead of `ctx.db.` - Correct?

**Reply by TheHinky** [reply]
I cant say for sure with drizzle, I assume likely would be similar with drizzle with similar size schema.

Yeah, sorry I should of stated, all I done was remove it from db (prisma in my case) from the context
https://github.com/t3-oss/create-t3-turbo/blob/main/packages/api/src/trpc.ts#L39

And yeah importing prisma directly, replace all ctx.prisma with prisma (and import). 

Might just be a recommendation for larger projects.
